lib/galaxy/config.py

```diff
@@ -363,9 +365,9 @@ def __init__( self, **kwargs ):
             # so dependending on the context a better default can be used (
             # request url in a web thread, Docker parent in IE stuff, etc...)
             galaxy_infrastructure_url = "http://localhost"
-            port = self.guess_galaxy_port()
-            if port:
-                galaxy_infrastructure_url += ":%s" % (port)
+            web_port = self.galaxy_infrastructure_web_port or self.guess_galaxy_port()
+            if web_port:
+                galaxy_infrastructure_url += ":%s" % (web_port)
```

Comment:
Okay, now this is all making sense. Thank you for the detailed explanation. 

I *want* to say "just sent a fake REMOTE_USER" header since that's what apache would do, but you really do need an externally accessible url because of the REMOTE_USER authorization header that I added to prevent doing exactly that. Hah.

I wish this was easily solveable with something like "oh, just partially fill out a URL in `galaxy_infrastructure_url`, just pass it `/` (or `/galaxy`) but since it's starting to be used in places that would have some horrendous effects I imagine. 

Okay, +1 from me, this is the best option for that case.